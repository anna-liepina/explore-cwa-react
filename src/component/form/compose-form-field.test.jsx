import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import compose from './compose-form-field';

const FormField = compose((props) => <input {...props} />);

describe('<FormField/>', () => {
    const props = {
    };

    const optionalProps = {
        'data-cy': 'optProps.data-cy',
        className: 'optProps.className',
        label: 'optProps.label',
        errors: ['optProps.errors[0]', 'optProps.errors[1]'],
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment } = render(<FormField {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props', () => {
            const { asFragment } = render(<FormField {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
