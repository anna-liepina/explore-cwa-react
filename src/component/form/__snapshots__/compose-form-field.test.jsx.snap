// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField/> render with default/required props 1`] = `
<DocumentFragment>
  <label
    class="form-input_label "
  >
    <input
      data-cy=""
    />
  </label>
</DocumentFragment>
`;

exports[`<FormField/> render with optional/required props 1`] = `
<DocumentFragment>
  <label
    class="form-input_label optProps.className"
  >
    optProps.label
    <input
      data-cy="optProps.data-cy"
    />
    <ul
      class="form-input_errors"
    >
      <li
        class="form-input_error"
        data-cy="optProps.data-cy-error-0"
      >
        optProps.errors[0]
      </li>
      <li
        class="form-input_error"
        data-cy="optProps.data-cy-error-1"
      >
        optProps.errors[1]
      </li>
    </ul>
  </label>
</DocumentFragment>
`;
